tree_segment.py

```diff
@@ -0,0 +1,434 @@
+import laspy
+import matplotlib.pyplot as plt
+import math
+import numpy as np
+import random
+import time
+import pandas as pd
+import argparse
+import re
+import os
+
+
+class Point_Cloud:
+    # object that stores the points in a point cloud, initialized from a las file
+    # points can be enabled or disabled, disabled points are ignored when searching for stems
+    def __init__(self, file_name, verbose=False):
+        all_data = laspy.read(file_name)
+        height_of_points_above_ground = np.array(
+            [point[-1] for point in all_data.points.array])
+        self.points = [Point(xyz, height) for xyz, height in zip(
+            all_data.xyz, height_of_points_above_ground)]
+        # internal, to access this use enabled_points()
+        self.__enabled_points__ = self.points
+        self.verbose = verbose
```

Comment:
Nit: Prefer logging through a standard library such as https://realpython.com/python-logging/

Meta Commentary:
This is squarely in the nice-to-have bucket, but is often less mental overhead as it's easier to spot code that is functional logic vs logging, and often integrates better with larger projects. (e.g. for big systems it's nice to be able to record all info logs *somewhere* but only look at warning and above by default).

Reply:
Ack! I was completely ignorant of logging, this does seem like a much better way to do levels of verboseness!